### DISCUSSION TITLE
The bug getting edge attributes based on edge ids

### DISCUSSION THREAD

**Discussion by w158937299** [JanusGraph/janusgraph#4371] [Q&A]
Version: 1.0.0

Storage Backend: inmemory

Mixed Index Backend: elasticresearch-8.6.0

API/Driver: Gremlin-python 3.2.6

Steps to Reproduce:

![image](https://github.com/JanusGraph/janusgraph/assets/49486148/bf807c9c-eaef-4a12-b307-459a39854c98)

As shown in this figure, Query 1 returned the IDs of all edges in the graph data, and Query 2 had both graph database engines attempt to fetch corresponding attributes based on the edge IDs they queried, but both returned empty sets. Query 3 returned the attributes of all existing edges in the graph data, proving that edge attributes do exist. Query 4 was able to correctly retrieve results for corresponding attributes based on node IDs. We found it strange that Janusgraph were unable to execute operations to query corresponding attributes based on edge IDs.

**Comment by li-boxuan** [comment]
<img width="783" alt="image" src="https://github.com/JanusGraph/janusgraph/assets/25746010/4a3326f4-077d-41d7-bfe8-ccbc0caaae31">

This does not look correct. JanusGraph edge IDs are strings like `557-388-9hx-3ag`, not numbers.

**Reply by li-boxuan** [reply]
<img width="783" alt="image" src="https://github.com/JanusGraph/janusgraph/assets/25746010/cb5912f4-1dc5-4861-9460-b4303779f805">
TinkerGraph, on the contrary, have numerical edge IDs. What you show here looks like JanusGraph edge ID.

**Reply by li-boxuan** [reply]
May I ask if this is a research project attempting to capture bugs of JanusGraph?

**Reply by w158937299** [reply]
Thank you for your reply, I'm sorry I wrote janusgraph and tinkergraph flags backwards. This is a research paper on detecting bugs in different graph databases.
